servers/zms/src/main/java/com/yahoo/athenz/zms/ZMSImpl.java

```diff
@@ -1568,6 +1583,38 @@ private void verifyNoServiceDependenciesOnDomain(String domainName, String calle
         }
     }
 
+    private void verifyServiceProvidersApproveDeletion(ResourceContext ctx, String auditRef, AthenzDomain domain, String caller) {
+
+        List<ServiceIdentity> servicesWithProviderEndpoint = domain.getServices()
```

Comment:
Not sure why we're looking at the services in the domain? The list of provider services that have an endpoint defined got nothing to do with the domain itself. The services are already defined in the service_providers role name in the sys.auth domain and we only need to pick the ones that have a provider endpoint set. So I'm expecting that when we call the ServiceProviderManager::refreshServiceProviders method to fetch the service providers, we also read each service identity and if the provider is set then we keep that in a separate set that we use here for the check.